sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2244 @@
+    }
+
+    struct PathCreateOptions
+    {
+      std::string ClientRequestId = std::string(); // Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the analytics logs when storage analytics logging is enabled.
+      uint32_t Timeout = uint32_t(); // The timeout parameter is expressed in seconds. For more information, see <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting Timeouts for Blob Service Operations.</a>
+      std::string ApiVersionParameter = k_DefaultServiceApiVersion; // Specifies the version of the operation to use for this request.
+      PathResourceType Resource = PathResourceType::Unknown; // Required only for Create File and Create Directory. The value must be "file" or "directory".
+      std::string Continuation = std::string(); // Optional.  When deleting a directory, the number of paths that are deleted with each invocation is limited.  If the number of paths to be deleted exceeds this limit, a continuation token is returned in this response header.  When a continuation token is returned in the response, it must be specified in a subsequent invocation of the delete operation to continue deleting the directory.
+      PathRenameMode Mode = PathRenameMode::Unknown; // Optional. Valid only when namespace is enabled. This parameter determines the behavior of the rename operation. The value must be "legacy" or "posix", and the default value will be "posix".
+      std::string CacheControl = std::string(); // Optional. Sets the blob's cache control. If specified, this property is stored with the blob and returned with a read request.
+      std::string ContentEncoding = std::string(); // Optional. Sets the blob's content encoding. If specified, this property is stored with the blob and returned with a read request.
+      std::string ContentLanguage = std::string(); // Optional. Set the blob's content language. If specified, this property is stored with the blob and returned with a read request.
+      std::string ContentDisposition = std::string(); // Optional. Sets the blob's Content-Disposition header.
+      std::string ContentType = std::string(); // Optional. Sets the blob's content type. If specified, this property is stored with the blob and returned with a read request.
+      std::string RenameSource = std::string(); // An optional file or directory to be renamed.  The value must have the following format: "/{filesystem}/{path}".  If "x-ms-properties" is specified, the properties will overwrite the existing properties; otherwise, the existing properties will be preserved. This value must be a URL percent-encoded string. Note that the string may only contain ASCII characters in the ISO-8859-1 character set.
+      std::string LeaseIdOptional = std::string(); // If specified, the operation only succeeds if the resource's lease is active and matches this ID.
```

Comment:
Is `LeaseIdOptional` also defined in swagger? Because obviously some of other options are also optional. For example, why is it not `CacheControlOptional`?

Reply:
[here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/Generated/DataLakeRestClient.cs#L1093) is `PathCreate` function in .Net SDK, they name it `LeaseId`.  [here](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/src/generated/src/models/index.ts#L429) and [here](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/src/generated/src/models/index.ts#L253) is in js sdk, they also name it `leaseId`. So maybe it's not part of swagger?

Reply:
There is a `LeaseIdRequired` comparing with the `LeaseIdOptional`. This is an alignment made with blob's swagger spec.

Reply:
fine